src/main/groovy/pl/allegro/tech/build/axion/release/domain/VersionResolver.groovy

```diff
@@ -44,27 +52,94 @@ class VersionResolver {
         GitRepository repository = (GitRepository) this.repository
         Pattern releaseTagPattern = ~/^${tagProperties.prefix}.*/
         Pattern nextVersionTagPattern = ~/.*${nextVersionProperties.suffix}$/
-
-        TagsOnCommit tags = repository.latestTags(releaseTagPattern)
-        TagsOnCommit previousTags = tags;
-
-        Map currentVersionInfo = versionFromTags(tags.tags, false, nextVersionTagPattern, versionFactory)
-        Version currentVersion = currentVersionInfo.version
-
-
+        
+        Map currentVersionInfo = null
+        Map previousVersionInfo = null
+        List<TagsOnCommit> allTaggedCommits = repository.allTaggedCommits(releaseTagPattern, null, true, true);
+        currentVersionInfo = versionFromTaggedCommits(allTaggedCommits, false, nextVersionTagPattern, versionFactory)
+        boolean isHead = currentVersionInfo.isHead
+        
+        
+        TagsOnCommit previousTags = repository.latestTags(releaseTagPattern)
```

Comment:
If I had a better idea of what was going on here, I could probably merge these two methods into one; however, I can't seem to replicate what's going on here and don't have the time to reverse engineer it.